### PR TITLE
feat: migrate Crud template to select widget and Multiselect v2

### LINKED ISSUE #10851: [Bug]: For select widget filterText in autocomplete is shown as array data type
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

When i use autocomplete for a select widget i see the filterText property as an array, but it should be a string. When I reference it, it works as a string but gives the wrong impression that it's an array

Screenshot 2022-02-03 at 12 18 24 PM

### Steps To Reproduce

1. Bind a select widget
2. Start typing .fi
3. Observe the type

@Tooluloope also check for multi-select, i didn't check that.

### Environment

Release

### Version

Cloud

### PR DESCRIPTION
**Pull request by Tooluloope**
## Description
This PR involves the following
1. Migrating DropDown widget to Select; continuation of #10297
2. Migrating all tests related to Dropdown to Select
3. updating CRUD generating template to use Select widget; updated via https://www.notion.so/appsmith/How-to-update-CRUD-template-09594645bf8e4077b4161c2ec095c27c
4. Fix auto-completion of Select widget #10851
5. Select Evaluation of `"1"` to number rather than string

Fixes  #10851

## Type of change

> Please delete options that are not relevant.

- Bugfix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

- Manual Testing
- Unit Test
- Cypress Test

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:red_circle: Total coverage has decreased</summary>

    // Code coverage diff between base branch:release and head branch: Feat/migrate-dropdown-to-select-widget 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :red_circle: | total | 55.73 **(-0.01)** | 37.05 **(0)** | 35.8 **(0)** | 56.1 **(0)**
 :red_circle: | app/client/src/utils/autocomplete/TernServer.ts | 52.71 **(-0.23)** | 40.83 **(-0.84)** | 36.21 **(0)** | 56.74 **(-0.25)**
 :green_circle: | app/client/src/widgets/SelectWidget/widget/index.tsx | 54.79 **(-0.14)** | 23.21 **(0.99)** | 50 **(0)** | 55.71 **(-0.17)**</details>

### CODE REVIEW

**Review by riodeuno — APPROVED** [review]
@Tooluloope Please confirm if this needs a migration. Also, the comment by @sbalaji1192 regarding code from another PR in this PR. We should merge the previous one, and find the differences not showing here, before merging this in.

**Comment by riodeuno** [comment]
@Tooluloope Let's update the description of this PR with the following details
- [ ] Where have we updated the name of the widget (suggested widgets, tests, etc) a list would be fine.
- [ ] The issue which was created associated with this PR. Let's link that issue.
- [ ] If we don't have any tests in this PR, why is that so? A small paragraph explaining why, would suffice.
- [ ] Type of change has two entries, can we have a description of which is a new feature and what is a bug fix?

**Comment by riodeuno** [comment]
@Tooluloope The jest tests are failing in this PR, which results in the build failure. Please look into that. https://github.com/appsmithorg/appsmith/runs/5190477527?check_suite_focus=true

**Comment by Tooluloope** [comment]
Hi @riodeuno. I'm planning to make this PR the Final Migration to https://github.com/appsmithorg/appsmith/pull/11082.

The Migration I mean here is just moving all Test old Dropdown Widget and MultiSelect Test cases to Select and MultiselectV2, also Move the usage of the Dropdown widget in Suggested Pane and Generate Crud Page to Select Widget.

I would mark this widget as a draft, and Sorry for making you review it when I thought I was done with the Other PR.

File: app/server/appsmith-server/src/main/java/com/appsmith/server/helpers/WidgetSuggestionHelper.java [hunk 1]
```diff
@@ -178,9 +178,9 @@ private static DataFields collectFieldsFromData(Iterator<Map.Entry<String, JsonN
         List<WidgetSuggestionDTO> widgetTypeList = new ArrayList<>();
         if(!fields.isEmpty()) {
             if(fields.size() < 2) {
-                widgetTypeList.add(getWidget(WidgetType.DROP_DOWN_WIDGET, fields.get(0), fields.get(0)));
+                widgetTypeList.add(getWidget(WidgetType.SELECT_WIDGET, fields.get(0), fields.get(0)));
             } else {
-                widgetTypeList.add(getWidget(WidgetType.DROP_DOWN_WIDGET, fields.get(0), fields.get(0)));
+                widgetTypeList.add(getWidget(WidgetType.SELECT_WIDGET, fields.get(0), fields.get(0)));
```
Comment (abhvsn) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/helpers/WidgetSuggestionHelper.java:183]: Not sure what's the use of `if-else` block here 🤔, can we remove this if not required?

**Review by abhvsn — APPROVED** [review]

**Pull request closed without merging**